daffodil-core/src/main/scala/org/apache/daffodil/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -103,6 +103,14 @@ trait ElementBaseGrammarMixin
 
   lazy val prefixedLengthElementDecl: PrefixLengthQuasiElementDecl = LV('prefixedLengthElementDecl){
     Assert.invariant(lengthKind == LengthKind.Prefixed)
+    optPrimType match {
+      case Some(primType) => primType match {
+        case PrimType.Integer | PrimType.NonNegativeInteger if lengthUnits == LengthUnits.Bits =>
```

Comment:
Is this check necessary here? I would expect a prefixed length quasi element would eventually end up hitting the other change below at some point?

Reply:
I think so. I wrote the test first to see if prefixed would work as expected with the other change and it didn't. My understanding is that there are two types here, the type of the element itself and the type of the prefix. This case checks the type of the element itself and the later check is for the prefix type.